src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -963,7 +963,16 @@ void DynamoView_KeyDown(object sender, KeyEventArgs e)
 
         void DynamoView_KeyUp(object sender, KeyEventArgs e)
         {
-            if (e.Key == Key.Escape)
+            if (e.Key == Key.Escape && dynamoViewModel.WatchEscapeIsDown)
+            {
+                dynamoViewModel.WatchEscapeIsDown = false;
+                dynamoViewModel.EscapeCommand.Execute(null);
+            }
```

Comment:
Just in case users press the ESC key when another window is active and then make the Dynamo window active and release the ESC key.